### PR TITLE
[Issue-3063] Merge initial state options

### PR DESCRIPTION
**Pull request by mbaxter**
## PR Description
Merge `--Xws-initial-state` weak subjectivity state argument with existing `--initial-state` argument. 

This modifies the startup behavior in a few ways:
* We'll now always parse the initial state if it is supplied.  When spinning up an existing database, we check it is consistent with the initial state.  We also use the initial state to modify weak subjectivity options.  
* Since we're always running validation against the initial state, when overriding the initial state for custom networks users will have to supply the initial state argument at each startup. 

Other changes:
* Parse `NetworkDefinition` in `NetworkOptions` to simplify access to the parsed value.

## Fixed Issue(s)
Part of #3063 

## Documentation
This PR expands the purpose of the `--initial-state` flag:  users can pass a genesis state as before, or a more recent state from which to sync.

### CODE REVIEW

**Comment by ajsutton** [comment]
Q before I've had a chance to look at the code: If initial state is always required, how do we handle networks that are still pre-genesis?

**Comment by mbaxter** [comment]
> Q before I've had a chance to look at the code: If initial state is always required, how do we handle networks that are still pre-genesis?

Initial state isn't required.  But if you're doing something like running `--network=medalla --initial-state=custom-genesis.ssz`, where you're overriding a built-in initial state, you'll have to always supply this initial state.

File: teku/src/main/java/tech/pegasys/teku/cli/converter/NetworkDefinitionConverter.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  public static final String CHECKPOINT_ERROR =
+      "Checkpoint arguments should be formatted as <BLOCK_ROOT>:<EPOCH_NUMBER> where blockRoot is a hex-encoded 32 byte value and epochNumber is a number in decimal format";
+
+  @Override
+  public NetworkDefinition convert(final String networkString) {
+    try {
+      checkNotNull(networkString);
+      return NetworkDefinition.fromCliArg(networkString);
+    } catch (Throwable e) {
+      throw new CommandLine.TypeConversionException(CHECKPOINT_ERROR);
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/converter/NetworkDefinitionConverter.java]: I don't understand how this error message lines up with the input.  The network definition name would be `medalla`  or a URL to a custom config file.  I didn't expect the ws checkpoint format to be valid here at all.

**Review by ajsutton — APPROVED** [review]
LGTM. I think the error message in `NetworkDefinitionConverter` just needs to be fixed.

File: teku/src/main/java/tech/pegasys/teku/cli/converter/NetworkDefinitionConverter.java (hunk 1 above)
Comment (mbaxter) [reply teku/src/main/java/tech/pegasys/teku/cli/converter/NetworkDefinitionConverter.java]: whoops - copy pasta

**Pull request merged**